test-case/check-alsabat.sh

```diff
@@ -64,11 +73,84 @@ function __upload_wav_file
     done
 }
 
+function set_pga_to_unity_gain ()
+{
+    tmp=$(amixer controls | grep "$pga" | grep Volume)
+    search="name="
+    cname=${tmp#*$search}
+
+    # Get volume min and step to compute value for cset
+    # for 0 dB gain. The amixer line looks like
+    # "| dBscale-min=-50.00dB,step=1.00dB,mute=1"
+    scale=$(amixer cget name="$cname" | grep "dBscale" || true)
+    search="dBscale-min="
+    tmp=${scale#*$search}
+    min_db="${tmp%%dB*}"
+    search="step="
+    tmp=${scale#*$search}
+    step_db="${tmp%%dB*}"
+
+    # Get multiplied by 100 values by removing decimal dot
+    min_x100="${min_db//.}"
+    step_x100="${step_db//.}"
+    val=$(printf %d "$(((-min_x100) / step_x100))")
+
+    # Apply the computed value for requested gain
+    amixer cset name="$cname" "$val"
+}
+
+function set_pgas_list_to_unity_gain ()
+{
+    if [[ -z "$1" ]]; then
+	return
+    fi
+
+    for pga in $1; do
```

Comment:
This is confusing, but let's try.

Reply:
Shellcheck complains about line `set_pgas_list_to_unity_gain $CAP_PGAS` with SC2086: Double quote to prevent globbing and word splitting. It works OK though.

Reply:
Yes, that's expected because you should ideally not do that either. You should ideally never use a single string to store multiple, whitespace-separated strings. My suggestion that you applied is only "the lesser of two evils", my suggestion was to do this bad thing only in a single place (here) keeping the `set_pgas_list_to_unity_gain()` function clean. Thanks for applying it.

So, ideally CAP_PGAS should be converted to an array. In practice we're going to assume that the output of `$TPLGREADER` is safe and you can tell shellcheck that with this:
```shell
CAP_PGAS=$($TPLGREADER "$tplg" -f "snd:$cap_snd" -d pga -v)
# Let's assume the output of $TPLGREADER is free of *, other special characters and spurious spaces.
# shellcheck disable=SC2086
set_pgas_list_to_unity_gain $CAP_PGAS
```